text/0095-cognito-construct-library/working-backwards-readme.md

```diff
@@ -0,0 +1,567 @@
+# Amazon Cognito Construct Library
+
+> Note to RFC reviewers: All comments in code blocks such as this one are either implementation notes, todos for
+> further investigation, or pieces of information that are important for this RFC review but will not be part of the
+> final README.
+
+[Amazon Cognito](https://docs.aws.amazon.com/cognito/latest/developerguide/what-is-amazon-cognito.html) provides
+authentication, authorization, and user management for your web and mobile apps. Your users can sign in directly with a
+user name and password, or through a third party such as Facebook, Amazon, Google or Apple.
+
+The two main components of Amazon Cognito are [user
+pools](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools.html) and [identity
+pools](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-identity.html). User pools are user directories
+that provide sign-up and sign-in options for your app users. Identity pools enable you to grant your users access to
+other AWS services.
+
+This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+
+## User Pools
+
+User pools allow creating and managing your own directory of users that can sign up and sign in. They enable easy
+integration with social identity providers such as Facebook, Google, Amazon, Microsoft Active Directory, etc. through
+SAML.
+
+Using the CDK, a new user pool can be created as part of the stack using the construct's constructor. You must specify
+the `userPoolName`, as so -
+
+```ts
+new UserPool(this, 'myuserpool', {
+  userPoolName: 'myawesomeapp-userpool',
+  // ...
+  // ...
+});
+```
+
+### Sign Up
+
+Users need to either be signed up by the app's administrators or can sign themselves up. This can be configured via the
+`selfSignUp` property under the `signUp` group.
+
+A welcome email and/or SMS can be configured to be sent automatically once user has signed up. This welcome email and/or
+SMS will carry the temporary password for the user. The user will use this password to log in and reset their password.
+The temporary password is valid only for a limited number of days.
+
+Consider the following code sample where a number of these properties are configured -
+
+```ts
+new UserPool(this, 'myuserpool', {
+  // ...
+  // ...
+  signUp: {
+    selfSignUp: true,
+    tempPasswordValidity: Duration.days(3),
+    welcomeEmail: {
```

Comment:
I don't think this should be a `welcomeEmail`. It's either an invitation e-mail (for administrator sign up with a temporary password) or a verification message (for self-signup and when attributes require validation). These are the terms used in the cognito console and in cloudformation (InviteMessageTemplate and VerificationMessageTemplate).

Reply:
It should also be configureable whether to confirm with [code or link](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cognito-userpool-verificationmessagetemplate.html#cfn-cognito-userpool-verificationmessagetemplate-defaultemailoption).

Reply:
You are right. Changes to this coming in the next update.